### DISCUSSION TITLE
Search bar field?

### DISCUSSION THREAD

**Discussion by younkim** [orchidsoftware/platform#2002] [Q&A] [answered]
I'm looking to write a search bar `Field`. I can sort of do it by `Group`ing it with three `Input`s like below,

![image](https://user-images.githubusercontent.com/8534941/145338832-72f485b5-cb03-48d7-ac48-597a0824931f.png)

Generated by 
```
public function fields(): array
{
    return [
        Group::make([
            Input::make('search_input')
                ->type('text')
                ->max(255)
                ->required()
                ->placeholder(__('search'))
                ->value($searchString),
            Button::make('Search')->method('searchProduct')->type(Color::SUCCESS()),
            Button::make('Clear')->method('clearSearch')->type(Color::SECONDARY()),
        ]),
    ];
}
````

But how do I make it look like this using the existing mechanism? I know `Layout::view('template')` is an option, but I'm looking to avoid directly placing Blade files.

![image](https://user-images.githubusercontent.com/8534941/145338811-cf7fd985-3bef-4611-b6fb-04ac2485c25b.png)

**Comment by tabuna** [comment, marked as answer]
The best way to do this is to use filters. You can find out about them on this page https://orchid.software/en/docs/filters/#eloquent-filter. If you need not hide them under the drop-down, then the selection layout has constants with templates.
